financial-services-accelerator/components/org.wso2.financial.services.accelerator.identity.extensions/src/main/java/org/wso2/financial/services/accelerator/identity/extensions/dcr/application/listener/ApplicationUpdaterImpl.java

```diff
@@ -0,0 +1,289 @@
+            throw new FinancialServicesException("Error while reading configured Identity providers", e);
+        }
+    }
+
+    public void setConditionalAuthScript(ServiceProvider serviceProvider,
+                                         LocalAndOutboundAuthenticationConfig localAndOutboundAuthenticationConfig)
+            throws FinancialServicesException {
+
+        try {
+            if (DCRUtils.getFapiCompliantPropertyFromSP(serviceProvider)) {
+                if (localAndOutboundAuthenticationConfig.getAuthenticationScriptConfig() == null) {
+                    TextFileReader textFileReader = TextFileReader.getInstance();
+                    String authScript = textFileReader.readFile
+                            (IdentityCommonConstants.CONDITIONAL_COMMON_AUTH_SCRIPT_FILE_NAME);
+                    if (StringUtils.isNotEmpty(authScript)) {
+                        AuthenticationScriptConfig scriptConfig = new AuthenticationScriptConfig();
+                        scriptConfig.setContent(authScript);
+                        scriptConfig.setEnabled(true);
+                        localAndOutboundAuthenticationConfig.setAuthenticationScriptConfig(scriptConfig);
+                    }
+                }
+            }
+        } catch (IdentityOAuthAdminException e) {
+            throw new FinancialServicesException("Error while retrieving the consumer application", e);
+        } catch (RequestObjectException e) {
+            throw new FinancialServicesException("Error while reading the service provider metadata", e);
+        } catch (IOException e) {
+            throw new FinancialServicesException("Error occurred while reading file", e);
+        }
+    }
+
+    public void doPreCreateApplication(ServiceProvider serviceProvider,
+                                       LocalAndOutboundAuthenticationConfig localAndOutboundAuthenticationConfig,
+                                       String tenantDomain, String userName) throws FinancialServicesException {
+
+    }
+
+    public void doPostCreateApplication(ServiceProvider serviceProvider,
+                                       LocalAndOutboundAuthenticationConfig localAndOutboundAuthenticationConfig,
+                                       String tenantDomain, String userName) throws FinancialServicesException {
+
+    }
+
+    public void doPostGetApplication(ServiceProvider serviceProvider, String applicationName,
+                                     String tenantDomain) throws FinancialServicesException {
+
+    }
+
+    public void doPreUpdateApplication(OAuthConsumerAppDTO oauthApplication,
+                                       ServiceProvider serviceProvider, LocalAndOutboundAuthenticationConfig
+                                               localAndOutboundAuthenticationConfig, String tenantDomain,
+                                       String userName) throws FinancialServicesException {
+
+        try {
+            boolean updateAuthenticator = false;
```

Comment:
Is there any way to identify  whether this `doPreUpdateApplication` is originated from DCR API? , or from updating UI authenticators?. 

This is to stop re setting authenticators when we update from the UI. 
( I think this issue still exists in OB3 as well )

Reply:
There isn't a way to identify that for now. But I am checking the whether LocalAndOutBoundAuthenticationConfig is null or Authentications steps are null and set accordingly